docs/source/daml/reference/packages.rst

```diff
@@ -12,75 +12,94 @@ This page gives reference information on DAML package dependencies:
+
+To do so, you first need to generate the DAML archive of ``foo``. Go into ``/home/user/foo`` and run ``daml build -o foo.dar``. This will create the DAML archive, ``/home/user/foo/foo.dar``.
+
+.. TODO (#4925): Make the above step redundant by letting users declare projects directly. Then update this doc.
+
+Next, we will update the project config for ``bar`` to use the generated DAML archive as a depndency. Go into ``/home/user/bar`` and change the ``dependencies`` field in ``daml.yaml`` to point to the created `DAML archive`:
 
 .. code-block:: yaml
 
-  sdk-version: 0.0.0
-  name: foo
-  source: daml
-  version: 1.0.0
-  exposed-modules:
-    - Some.Module
-    - Some.Other.Module
   dependencies:
     - daml-prim
     - daml-stdlib
-    - /home/johndoe/bar/.daml/dist/bar-1.0.0.dar
+    - ../foo/foo.dar
 
-The import path needs to be the relative or absolute path pointing to the created DAML archive of
-the ``bar`` project. The archive can reside anywhere on the local file system. Note that the SDK
-versions of the packages ``foo`` and ``bar`` need to match, i.e. it is an error to import a package
-that was created with an older SDK.
+The import path can also be absolute, for example, by changing the last line to:
 
-Once a package has been added to the dependencies of the ``foo`` project, modules of ``bar`` can be
-imported as usual with the ``import Some.Module`` directive (see :ref:`Imports <daml-ref-imports>`).
-If both projects ``foo`` as well as ``bar`` contain a module with the same name, the import can be
-disambiguated by adding the package name in front of the module name, e.g. ``import "bar"
-Some.Module``.
+.. code-block:: yaml
 
-Note that all modules of package ``foo`` that should be available as imports of other packages need
-to be exposed by adding them to the ``exposed-modules`` stanza of the `daml.yaml` file. If the
-``exposed-modules`` stanza is omitted, all modules of the project are exposed by default.
+    - /home/user/foo/foo.dar
 
-Importing archives compiled with different SDK's
-************************************************
+When you run ``daml build`` in ``bar`` project, the compiler will make the definitions in ``foo.dar`` available for importing. For example, if ``foo`` exports the module ``Foo``, you can import it in the usual way:
 
-All DAML archive dependencies of a project need to be compiled with the same SDK as the project
-itself. However, it is possible to import templates and data types of an archive compiled with an
-older SDK, by listing them under the ``data-dependencies`` stanza:
+.. code-block:: daml
 
-.. code-block:: yaml
+  import Foo
+
+Sometimes you will have multiple packages with the same exported module name. In that case, you must also specify which package the module comes from, as follows:
+
+.. code-block:: daml
+
+  import "foo" Foo
```

Comment:
We should document how you can rename modules if you have the same name but different versions so you cannot use package-qualified imports to disambiguate. Totally fine to do that in a separate PR. This is also not specific to dependencies or data-dependencies so I wonder if it would make sense to put this into a separate “Handling colliding module names” section at the end or somethin glike that.

Reply:
is this just `import "foo" Foo as Foo1`, or something else? I agreed that dealing with conflicting module names should move to a separate section.

Reply:
No you need to specify something like `--package foo-1.0.0 (Foo as Foo1)` in your `daml.yaml` in that case.

Reply:
See https://github.com/digital-asset/daml/blob/master/compiler/damlc/tests/src/DA/Test/Packaging.hs#L102

Reply:
Oh, I see, this is for the case where you have two packages with the same name but different versions. This is terrible UX. Could we not change unit IDs to make package qualified imports work in this case?

Reply:
(I mean, change the way unit IDs are chosen from LF packages, so they include the package version.)

Reply:
GHC has fairly strictly rules about what can be a package name sadly. `foobar-1.0.0` is a valid unit id but it’s not a valid package name (the latter is what you use in package-qualified imports). I believe `foobar-1_0_0` would be but that doesn’t seem very satisfying. I looked into patching GHC at some point to allow the unit id in package-qualified imports and it looked fairly doable but then decided that I’m too lazy :slightly_smiling_face: I think for now, documenting the current state first and then seeing if we can improve upon it seems like the best option.

Reply:
Sounds good 👍 

Maybe in the future the user could specify an alias for packages in `dependencies` or `data-dependencies`, e.g.

``` yaml
dependencies:
- daml-prim
- daml-stdlib
- package: foo-1.4.2.dar
  alias: foo-v1  
```

or whatever syntax is most appropriate

Reply:
ah yeah, I thought about this in the past but then forgot about it. That sounds like a reasonably simple option :+1:

Reply:
I added an issue so we don't forget about it:
https://github.com/digital-asset/daml/issues/4948